archinfo/arch.py

```diff
@@ -91,11 +93,15 @@ class Arch(object):
     byte_width = 8
 
     def __init__(self, endness):
-        if endness not in (Endness.LE, Endness.BE, Endness.ME):
-            raise ArchError('Must pass a valid VEX endness: Endness.LE or Endness.BE')
+        if self.vex_support:
+            if endness not in (Endness.LE, Endness.BE, Endness.ME):
+                raise ArchError('Must pass a valid VEX endness: Endness.LE or Endness.BE')
```

Comment:
endness is supposed to be independent from VEX, that's why we have the enum... Can you make this a better condition? has_memory?

Reply:
has_memory can be inferred from memory_endness is None, like the below?